zerver/lib/home.py

```diff
@@ -111,27 +111,42 @@ def build_page_params_for_home_page_load(
     first_in_realm: bool,
     prompt_for_invites: bool,
     needs_tutorial: bool,
+    realm: Realm,
 ) -> Tuple[int, Dict[str, Any]]:
     """
     This function computes page_params for when we load the home page.
 
     The page_params data structure gets sent to the client.
     """
-    client_capabilities = {
-        "notification_settings_null": True,
-        "bulk_message_deletion": True,
-        "user_avatar_url_field_optional": True,
-    }
-
-    register_ret = do_events_register(
-        user_profile,
-        request.client,
-        apply_markdown=True,
-        client_gravatar=True,
-        slim_presence=True,
-        client_capabilities=client_capabilities,
-        narrow=narrow,
-    )
+
+    if user_profile is not None:
+        client_capabilities = {
+            "notification_settings_null": False,
+            "bulk_message_deletion": True,
+            "user_avatar_url_field_optional": True,
+        }
```

Comment:
I think we want `client_capabilities` outside the conditional block, structurally;